tests/test_ls.py

```diff
@@ -39,6 +39,48 @@ def test_ls_caching(fs: LakeFSFileSystem, repository: Repository) -> None:
     assert counter.count("objects_api.list_objects") == 1
 
 
+def test_ls_with_one_dir(
+    fs: LakeFSFileSystem,
+    repository: Repository,
+    temp_branch: Branch,
+) -> None:
+    """
+    Check that ls calls are properly cached.
+    """
+    root = f"lakefs://{repository.id}/{temp_branch.id}"
+
+    fs.mkdir(f"{root}/test")
+    fs.cp(f"{root}/lakes.parquet", f"{root}/test/lakes.parquet")
+    fs.cp(f"{root}/lakes.parquet", f"{root}/test/lakes2.parquet")
+    fs.rm(f"{root}/README.md")
+    fs.rm(f"{root}/data/", recursive=True)
+    fs.rm(f"{root}/images/", recursive=True)
+    fs.rm(f"{root}/lakes.parquet")
+
+    # Check root
+    root_resource = f"{root}/"
+    list_of_files = fs.ls(root_resource)
+    assert [fs.unstrip_protocol(x["name"]) for x in list_of_files] == [f"{root_resource}test/"]
+    assert len(fs.dircache) == 1
+    assert set(map(lambda x: fs.unstrip_protocol(x), fs.dircache.keys())) == {
+        root_resource.removesuffix("/")
+    }
```

Comment:
For consistent style (the rest of the file uses comprehensions instead of `map`):
```suggestion
    assert {fs.unstrip_protocol(x) for x in fs.dircache.keys()} == {root_resource.removesuffix("/")}
```